src/java/org/apache/cassandra/db/compaction/CompactionPick.java

```diff
@@ -58,9 +64,8 @@
     /** The total size on disk for the sstables in this compaction */
     private final long totSizeInBytes;
 
-    /** The unique compaction id, this is only available when a compaction is submitted */
-    @Nullable
-    private volatile UUID id;
+    /** This is set to true when the compaction is submitted */
+    private volatile boolean submitted;
```

Comment:
Nit: Maybe combine `submitted` and `completed` into a single enum value to clarify the state progression?

Reply:
I guess it makes sense to make such change together with improving usage of the variables. Currently `completed` is not used anywhere to my understanding except setting it.